### ISSUE TITLE
How do the permissions and roles work?

### ISSUE THREAD

**Issue by mwo44** [pengkong/A3M-for-CodeIgniter-2.0#32]
Hi
first of all I’d like to thank you for the great piece of code and functionality. It’s my first use of it.
And now the question: how do the permissions and roles work?
When I look at the database structure I see there are tables like a3m_acl_permission and a3m_acl_role
now does the library support these functionalities? I managed to install and run the library. Tested it, and started creating views for non-logged-in and logged-in users. But now I see I will need some more complexity.
Basically what I need is to divide my users into readers and publishers. Then I will have at least two categories of publishers: these who can write and these who can moderate (higher permission level). How can I achieve that? User management doesn’t seem to be so simple as I thought at the beginning :-(

Once again, this library is great.
Thanks in advance

**Comment by donjakobo** [comment]
As I no longer support this branch of a3m, take a look at the latest fork (and branch w the admin panel in beta): https://github.com/donjakobo/A3M/tree/admin-panel

**Comment by mwo44** [comment]
Ok, I will. Thank you
